bpfd/src/bpf.rs

```diff
@@ -772,6 +767,26 @@ impl BpfManager {
             .collect()
     }
 
+    pub(crate) fn get_program(&mut self, id: u32) -> Result<Program, BpfdError> {
+        debug!("BpfManager::get_program({0})", id);
+        // If the program was loaded by bpfd, then us it.
+        // Otherwise, call Aya to get ALL the loaded eBPF programs, and convert the data
+        // returned from Aya into an Unsupported Program Object.
+        match self.programs.get(&id) {
+            Some(p) => Ok(p.to_owned()),
+            None => loaded_programs()
+                .find_map(|p| {
+                    let prog = p.map_err(BpfdError::BpfProgramError).ok()?;
```

Comment:
No need to map the error if we're just discarding it :) 
```suggestion
                    let prog = p.ok()?;
```